keel-sql/keel-sql.gradle.kts

```diff
@@ -65,39 +66,50 @@ dependencies {
   liquibaseRuntime("mysql:mysql-connector-java")
 }
 
-tasks.register<Exec>("runMysql") {
-  commandLine("docker run --name mysqlJooq -d --rm -e MYSQL_ROOT_PASSWORD=sa -e MYSQL_DATABASE=keel -p 6603:3306 mysql:5.7".split(" "))
-  doLast {
-    // Wait for the DB server to come up...
-    sleep(15 * 1000)
+// expand properties in jooqConfig.xml so it gets a fully-qualified directory to generate into
+tasks.withType<ProcessResources> {
+  filesMatching("jooqConfig.xml") {
+    expand(project.properties)
   }
 }
 
-tasks.register<Exec>("stopMysql") {
-  commandLine("docker stop mysqlJooq".split(" "))
+tasks.getByName<LiquibaseTask>("liquibaseUpdate") {
+  inputs.dir("$projectDir/src/main/resources/db")
+  outputs.dir("$projectDir/src/generated/java")
+
+  doFirst {
+    if (!buildingInDocker) {
+      exec {
+        commandLine("sh", "-c", "for PS in \$(docker ps -q --filter name=mysqlJooq); do docker stop \$PS; done")
+      }
+      exec {
+        commandLine("sh", "-c", "docker run --name mysqlJooq --health-cmd='mysqladmin ping -s' -d --rm -e MYSQL_ROOT_PASSWORD=sa -e MYSQL_DATABASE=keel -p 6603:3306 mysql:5.7; while STATUS=\$(docker inspect --format \"{{.State.Health.Status}}\" mysqlJooq); [ \$STATUS != \"healthy\" ]; do if [ \$STATUS = \"unhealthy\" ]; then echo \"Docker failed to start\"; exit -1; fi; sleep 1; done")
```

Comment:
nit: Consider breaking the script in multiple lines to make it a bit more readable. It's tough to parse through the one-liner.